prompt/prompt-executor/prompt-executor-clients/prompt-executor-anthropic-client/src/commonMain/kotlin/ai/koog/prompt/executor/clients/anthropic/AnthropicLLMClient.kt

```diff
@@ -156,30 +149,22 @@ public open class AnthropicLLMClient(
 
         val request = createAnthropicRequest(prompt, tools, model, false)
 
-        return withContext(Dispatchers.SuitableForIO) {
-            val response = httpClient.post(DEFAULT_MESSAGE_PATH) {
-                setBody(request)
-            }
-
-            if (response.status.isSuccess()) {
-                val anthropicResponse = response.body<AnthropicResponse>()
-                processAnthropicResponse(anthropicResponse)
-            } else {
-                // TODO: after the update to the KoogHttpClient, delegate this logic to the http client
-
-                val httpClientException = KoogHttpClientException(
-                    statusCode = response.status.value,
-                    errorBody = response.bodyAsText(),
-                )
-                val exception = LLMClientException(
-                    clientName = clientName,
-                    message = httpClientException.message,
-                    cause = httpClientException
-                )
-                logger.error(exception) { exception.message }
-                throw exception
-            }
-        }
+        return try {
+            httpClient.post(
+                path = settings.messagesPath,
+                request = request,
+                requestBodyType = String::class,
+                responseType = AnthropicResponse::class,
```

Comment:
This isn’t directly related to this PR, but it makes me wonder why we explicitly specify the types.
Why not do it the same way as Ktor requests and use reified types instead?

Reply:
I think we just wrap it into exception inside ktor `KoogHttpCLient`, all should be good there